client/sites-dashboard/components/sites-site-url.tsx

```diff
@@ -0,0 +1,31 @@
+import styled from '@emotion/styled';
+import { ReactNode, useState, AnchorHTMLAttributes } from 'react';
+import ExternalLink from 'calypso/components/external-link';
+
+const ExternalSiteLink = styled( ExternalLink )`
+	text-overflow: ellipsis;
+	overflow: hidden;
+	display: inline-block;
+	font-size: 14px;
+	color: var( --studio-gray-60 ) !important;
+
+	&:hover,
+	&:visited {
+		font-weight: 500;
+	}
+`;
+
+interface ExternalSiteLinkProps extends AnchorHTMLAttributes< HTMLAnchorElement > {
+	children?: ReactNode;
+}
+
+export const SiteUrl = ( { children, ...props }: ExternalSiteLinkProps ) => {
+	const [ icon, setIcon ] = useState( false );
+	return (
+		<div onMouseEnter={ () => setIcon( true ) } onMouseLeave={ () => setIcon( false ) }>
```

Comment:
I'm still not a huge fan of using JavaScript events for basic `:hover` behavior...

Is there no way to use `:hover` CSS in whatever version of the React paradigm we're using?

Reply:
The thing is that it's not simply styling, right? We're effectively adding and removing elements, it's a matter of logic, not presentation.

I tend to think that adding or removing elements through CSS is _actually the hack_ and not the best solution. CSS should care about styling, not business logic.

And to top it, an element should definitely not interfere with how other elements are displayed (or not).

Reply:
> We're effectively adding and removing elements, it's a matter of logic, not presentation.

@zaguiini Oh, that's a good point. I suppose you're right. I guess it also illustrates the conceptual complexity of showing/hiding elements based on mouse events.

Reply:
I tend to agree that we shouldn't be using JavaScript to implement `:hover` behaviour, that's what the `:hover` selector is for after all! But I'm also wondering whether the "only show the icon on hover" idea has some issues. When I talked about it with @SaxonF we were talking about trying it out just to see what it looks like. But with long URLs it can make the layout jump around. I'll record a video.

Reply:
I think the long url should be clipped, not sure why it's not clipped here. Will check to see where the style is being overwritten. As for the approach I have no issue using the css hover on child icon to hide or show it, probably in this case it's more expensive to call react rendering on mouse effect as oppose to the css `:hover`. If we go css route will it affect the reusability of the component? It seems `site-url` is fixed to sites-dashboard for now until we see a need to use it elsewhere so if css is simpler and less expensive I'd rather that route.

Reply:
That's OK too.